app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicsListFragment.kt

```diff
@@ -1,20 +1,27 @@
 package com.worldsnas.starplayer.view.musics_list
 
 import android.os.Bundle
-import androidx.fragment.app.Fragment
+import android.util.Log
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.Observer
 import com.worldsnas.starplayer.App
-
-import com.worldsnas.starplayer.R
 import com.worldsnas.starplayer.databinding.FragmentMusicsListBinding
 import com.worldsnas.starplayer.di.DaggerMusicListComponent
 import com.worldsnas.starplayer.di.MusicListComponent
 import com.worldsnas.starplayer.model.Music
+import com.worldsnas.starplayer.view.ViewModelFactory
+import javax.inject.Inject
 
 class MusicsListFragment : Fragment() {
 
+    @Inject
+    lateinit var viewModelFactory: ViewModelFactory
+    private val musicListViewModel by lazy {
```

Comment:
az property delegate ha ye jadid ketabkhone ViewModel estefade konid

Reply:
age emkanesh hast ino bishtar tozih bedid, nmidonm chi hastand

Reply:
ye extension delegate e  be name:
`viewModels`
berid dakhele safehe doc ViewModel mibinid